apps/meteor/client/views/setupWizard/steps/RegisterServerStep.tsx

```diff
@@ -1,57 +1,80 @@
 import type { ReactElement, ComponentProps } from 'react';
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 
+import { queryClient } from '../../../lib/queryClient';
+import { dispatchToastMessage } from '../../../lib/toast';
 import { useSetupWizardContext } from '../contexts/SetupWizardContext';
 
 const SERVER_OPTIONS = {
 	REGISTERED: 'REGISTERED',
-	STANDALONE: 'STANDALONE',
+	OFFLINE: 'OFFLINE',
 };
 
 const RegisterServerStep = (): ReactElement => {
-	const { goToPreviousStep, currentStep, setSetupWizardData, registerServer, maxSteps, offline, completeSetupWizard } =
+	const t = useTranslation();
+	const { currentStep, goToNextStep, setSetupWizardData, registerServer, maxSteps, offline, completeSetupWizard, registerPreIntent } =
 		useSetupWizardContext();
 	const [serverOption, setServerOption] = useState(SERVER_OPTIONS.REGISTERED);
 
-	const router = useRoute('cloud');
-
-	const handleRegisterOffline: ComponentProps<typeof RegisterServerPage>['onSubmit'] = async () => {
-		await completeSetupWizard();
-		router.push({}, { register: 'true' });
+	const handleRegister: ComponentProps<typeof RegisterServerPage>['onSubmit'] = async (data: {
+		email: string;
+		agreement: boolean;
+		updates: boolean;
+		resend?: boolean;
+	}) => {
+		goToNextStep();
+		setSetupWizardData((prevState) => ({ ...prevState, serverData: data }));
+		await registerServer(data);
 	};
 
-	const handleRegister: ComponentProps<typeof RegisterServerPage>['onSubmit'] = async (data) => {
-		if (data.registerType !== 'standalone') {
-			setSetupWizardData((prevState) => ({ ...prevState, serverData: data }));
-			await registerServer(data);
-		}
-	};
+	const [clientKey, setClientKey] = useState('');
+
+	const registerManually = useEndpoint('POST', '/v1/cloud.manualRegister');
+	const getWorkspaceRegisterData = useMethod('cloud:getWorkspaceRegisterData');
+
+	useEffect(() => {
+		const loadWorkspaceRegisterData = async (): Promise<void> => {
+			const clientKey = await getWorkspaceRegisterData();
+			setClientKey(clientKey);
+		};
+
+		loadWorkspaceRegisterData();
+	}, [getWorkspaceRegisterData]);
+
+	useEffect(() => {
+		registerPreIntent();
+	}, [registerPreIntent]);
+
+	const handleConfirmOffline: ComponentProps<typeof RegisterOfflinePage>['onSubmit'] = async ({ token }) => {
+		try {
+			await registerManually({ cloudBlob: token });
+			queryClient.invalidateQueries(['licenses']);
 
-	const handleConfirmStandalone: ComponentProps<typeof StandaloneServerPage>['onSubmit'] = async ({ registerType }) => {
-		if (registerType !== 'registered') {
 			return completeSetupWizard();
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: t('Cloud_register_error') });
 		}
 	};
 
-	if (serverOption === SERVER_OPTIONS.STANDALONE) {
+	if (serverOption === SERVER_OPTIONS.OFFLINE) {
 		return (
-			<StandaloneServerPage
-				currentStep={currentStep}
+			<RegisterOfflinePage
+				termsHref=''
+				policyHref=''
```

Comment:
Is this correct? Don't we have a hrefs for terms and policy? Since this is Offline registration I feel this could be important

Reply:
No need to pass terms and policy, removing it!